addons/account/models/partner.py

```diff
@@ -249,17 +249,35 @@ def _get_fiscal_position(self, partner, delivery=None):
         if not partner:
             return self.env['account.fiscal.position']
 
-        company = self.env.company
-        intra_eu = vat_exclusion = False
-        if company.vat and partner.vat:
-            eu_country_codes = set(self.env.ref('base.europe').country_ids.mapped('code'))
-            intra_eu = company.vat[:2] in eu_country_codes and partner.vat[:2] in eu_country_codes
-            vat_exclusion = company.vat[:2] == partner.vat[:2]
-
-        # If company and partner have the same vat prefix (and are both within the EU), use invoicing
-        if not delivery or (intra_eu and vat_exclusion):
+        # If no "delivery" partner is specified, we assume it will be the "invoicing" partner.
+        if not delivery:
             delivery = partner
 
+        company = self.env.company
+        eu_country_codes = set(self.env.ref('base.europe').country_ids.mapped('code'))
+        delivery_country = delivery.country_id
+        vat_valid = self._get_vat_valid(partner, company)
+
+        eu_vat_partner = partner.vat and partner.vat[:2] in eu_country_codes
+        eu_partner = partner.country_code in eu_country_codes
+        eu_delivery = delivery.country_code in eu_country_codes
+        domestic_delivery = delivery_country == company.country_id
+        external_delivery = delivery_country != partner.country_id
+
+        # If the delivery is to a different country than the partner's country (external delivery),
+        # the delivery is within the EU, and the partner does not have a valid EU VAT number,
+        # then assign the company's country as the delivery country) and force vat_valid to True
+        # to get the domestic fp
+        if external_delivery and eu_delivery and not eu_vat_partner:
+            delivery_country = company.country_id
+            vat_valid = True
+
+        # If the delivery is to the same country than the company's country (domestic delivery),
+        # the partner has a valid EU VAT number but is not from EU
+        # We need to force vat_valid to False in order to get the EU private fp
```

Comment:
```suggestion
        # the partner has a valid EU VAT number that is not from EU,
        # we need to force vat_valid to False in order to get the EU private FP
```

Reply:
but is the word I want to use, partner has a valid EU VAT, but he is not from EU ;-)